byway/src/main/webapp/script.js

```diff
@@ -15,14 +15,144 @@
+    zoom: 13,
+    mapTypeId: "roadmap"
+  });
+
+
+  if (navigator.geolocation) {
+    navigator.geolocation.getCurrentPosition(function(position) {
+      userlatlng = {
+        lat: position.coords.latitude,
+        lng: position.coords.longitude
+      };
+      map.setCenter(userlatlng);
+    }, function() {
+         map.setCenter(defaultCenter);
+        });
+  } 
+  else {
+    window.alert("Geolocation failed");
+  }
+  // Create the search boxes and link them to the UI elements.
+  createSearchBox(map,'pac-input');
+  createSearchBox(map,'pac-input-2');
+  
+}
+
+function createSearchBox(map,container){
+  const start = document.getElementById(container);
+  const searchBox = new google.maps.places.SearchBox(start);
+
+   map.addListener("bounds_changed", () => {
+    searchBox.setBounds(map.getBounds());
+  });
+  let markers = [];
+
+  searchBox.addListener("places_changed", () => {
+    const places = searchBox.getPlaces();
+
+    if (places.length == 0) {
+      return;
+    }
+    // Clear out the old markers.
+    markers.forEach(marker => {
+      marker.setMap(null);
+    });
+    markers = [];
+    // For each place, get the icon, name and location.
+    const bounds = new google.maps.LatLngBounds();
+    places.forEach(place => {
+      if (!place.geometry) {
+        console.log("Returned place contains no geometry");
+        return;
+      }
+      const icon = {
+        url: place.icon,
+        size: new google.maps.Size(71, 71),
+        origin: new google.maps.Point(0, 0),
+        anchor: new google.maps.Point(17, 34),
+        scaledSize: new google.maps.Size(25, 25)
+      };
+      // Create a marker for each place.
+      markers.push(
+        new google.maps.Marker({
+          map,
+          icon,
+          title: place.name,
+          position: place.geometry.location
+        })
+      );
+
+      if (place.geometry.viewport) {
+        // Only geocodes have viewport.
+        bounds.union(place.geometry.viewport);
+      } else {
+        bounds.extend(place.geometry.location);
+      }
+    });
+    map.fitBounds(bounds);
+  });
+
+}
+
+/* fetches start location and destinations from DestinationsServlet and adds to DOM*/
+function getLocations(){
+  fetch('/destinations').then(response => response.json()).then((userLocations) => {
+    document.getElementById('start-location').innerText = "Start Location :" + userLocations.start;
+    const container = document.getElementById('destinations-container');
+    container.innerText = "Destinations:";
+    let destinationArray= userLocations.destinations;
+    destinationArray.forEach((destination) => {
+      let destinationToAdd = document.createElement('p');
+      destinationToAdd.innerText = destination;
+      container.appendChild(destinationToAdd);
+    }) 
+  });
+}
+
+/* fills Start location Searchbox with previously input */
+function getStartDestination(){
+  fetch('/destinations').then(response => response.json()).then((userLocations) =>{
```

Comment:
Note that each time you call `fetch` you are sending another request to the server, which takes time. You may want to find a way to store the response and re-use it, or at least use only one fetch for multiple setup steps, for example, having one method like: `function fetchDestinations()` that returns `fetch('/destinations').then(response => response.json())`, and use the result of that for both `setStartDestination` and `setLocations`.

In general it would be a good idea to try to separate your *data* from how that data is *presented* to the user; if you always just write data to the page, most of your data exists only in the HTML and is difficult to retrieve again/hard to reason about. Storing data in JS and then just using HTML to present that data makes the data easier to reason about, because it has a known location and a consistent structure, and it makes the page easier to reason about and keep properly in sync, because the page is nothing more than a way of presenting a specific portion of the data.

Reply:
added a functions called fetchDestinaitons()

Reply:
I wasn't able to get to this either, it'll be done Monday!

Reply:
done